packages/derive/src/callable_point.rs

```diff
@@ -69,3 +69,61 @@ fn make_call_origin_and_return(
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use syn::{parse_quote, ItemFn};
+
+    const PART_CALL_TO_ORIGIN_ARG0: &str = "foo () ;";
+    const PART_CALL_TO_VEC: &str = "cosmwasm_std :: to_vec (";
+    const PART_CALL_TO_RELEASE_BUFFER: &str = "cosmwasm_std :: memory :: release_buffer (";
+
+    #[test]
+    fn make_call_origin_and_return_works() {
+        {
+            let function_foo_ret1: ItemFn = parse_quote! {
+                fn foo() -> u64 {
+                    1
+                }
+            };
+            /* generated:
+            let result = foo () ;
+            let vec_result = cosmwasm_std :: to_vec (& result) . unwrap () ;
+            cosmwasm_std :: memory :: release_buffer (vec_result) as u32
+            */
+            let result_code = make_call_origin_and_return(
+                &function_foo_ret1.sig.ident,
+                0,
+                &function_foo_ret1.sig.output,
+            )
+            .to_string();
+            assert_eq!(result_code.matches(PART_CALL_TO_ORIGIN_ARG0).count(), 1);
+            assert_eq!(result_code.matches(PART_CALL_TO_VEC).count(), 1);
+            assert_eq!(result_code.matches(PART_CALL_TO_RELEASE_BUFFER).count(), 1);
```

Comment:
```rust
            let expected: TokenStream = parse_quote! {
                let result = foo();
                let vec_result = cosmwasm_std::to_vec(&result).unwrap();
                cosmwasm_std::memory::release_buffer(vec_result) as u32
            };
            ...
            assert_eq!(expected.to_string(), result_code)
```

This test works.
I think this test covers more than now.

Reply:
Great! thks for your suggestion. I didn't like the existing test code style. It's more looks good.
I replaced all.